src/Silex/Provider/SecurityServiceProvider.php

```diff
@@ -140,14 +142,20 @@ public function register(Container $app)
             return new UserChecker();
         };
 
-        $app['security.access_manager'] = function ($app) {
-            $votersIterator = new ServiceIterator($app, $app['security.voter_services']);
+        $app['security.access_manager'] = function ($app) use ($default_voters) {
+            $voter_services = $app['security.voter_services'];
+            if (empty(array_diff($voter_services, $default_voters))) {
+                // BC: If you don't need lazy voter service evaluation, you could still extend security.voters
```

Comment:
we need to trigger a deprecation warning if they extend stuff in the old way rather than the new one, to tell them to migrate

Reply:
@stof I'm not sure extending `security.voters` should be deprecated though unless we want to move to lazy-only. I mean you may not have a circular reference dependency in your voter.

Reply:
Well, having 2 different ways to register voters (especially when they cannot work in parallel, as was the case at the time of my previous review) is definitely a bad DX.

Reply:
We can make it work in parallel if we register services on the fly based on the voters objects and extend the services with them, but this seems quite hacky.